third_party/nvfuser/csrc/expr_simplifier.cpp

```diff
@@ -1271,6 +1271,18 @@ bool lessThan(Val* x, Val* y, const Context& context) {
       return true;
     }
   }
+  for (const auto& [a, b] : context.getKnownLessThan()) {
+    // x < b & b <= y  -->  x < y
+    if (a->sameAs(x) && lessEqual(b, y, context)) {
```

Comment:
Why not `lessEqual(x, a, context)` instead of `a->sameAs(x)`? If `x <= a`, `a < b` and `b <= y`, then `x < y`, right?

Reply:
I agree, will change

Reply:
Why infinite recursion?

Reply:
Looks like if I only touch `lessThan`, there won't be infinite recursion.  Infinite recursion will happen if I also changed `lessEqual`, because if  I assume `i1 < i2` and want to prove `x < y`, the I will end up trying to prove `x <= i1` forever. Updated https://github.com/NVIDIA/Fuser/pull/34

Reply:
Ah, OK, yeah, in that case I see that could happen.